app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,23 @@
+class CommentsController < ApplicationController
+  def create
+    @user = User.find(params[:user_id])
+    @post = Post.find(params[:post_id])
+
+    @comment = Comment.new(comment_params)
+    @comment.author = @user
+    @comment.post = @post
+
+    if @comment.save
+      redirect_to user_post_url({ author_id: @user.id, id: @post.id })
+      flash[:success] = 'Comment created successfully'
+    else
+      render :new, flash: { error: 'Please make sure your post is a valid comment' }
+    end
+  end
```

Comment:
It will be nice to refactor your create action to have a new action method of it's own. Like so:

```ruby
def new
  @comment = Comment.new
end
```
Reason as given above.